src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessCollectionTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Diagnostics.ProcessTests
+{
+    public class ProcessCollectionTests : ProcessTestBase
+    {
+        [Fact]
+        public void TestModuleCollectionBehavior()
+        {
+            ProcessModule[] mArray = _process.Modules.Cast<ProcessModule>().ToArray();
+
+            ProcessModuleCollection moduleCollection = new ProcessModuleCollection(mArray);
+            Assert.True(moduleCollection.Count > 1);
+            Assert.Equal(mArray[0], moduleCollection[0]);
+            Assert.True(moduleCollection.Contains(mArray[0]));
+
+            ProcessModule[] moduleArray = new ProcessModule[moduleCollection.Count + 1];
+            moduleCollection.CopyTo(moduleArray, 1);
+            Assert.Equal(mArray[0], moduleArray[1]);
+
+            Assert.Equal(0, moduleCollection.IndexOf(mArray[0]));
+        }
```

Comment:
> I'll add tests for explicit interface implementations as well.

Great, thanks.

> InnerList and set_Item are not exposed in ProcessModuleCollection api contract.

Good to know.  InnerList should be changed to be private rather than protected, right?